pkg/registry/registry.go

```diff
@@ -156,3 +159,47 @@ func (r *Registry) DeleteImage(image string, tag string) error {
 func (r *Registry) RetaggedName(image string) string {
 	return images.RetaggedName(r.host, r.organisation, image)
 }
+
+// GuessRegistryPath examines the given image string,
+// determines whether it describes a full image path,
+// is hosted on Docker hub, or belongs to the Docker standard library, and returns a URL representing the full path.
+func (r *Registry) GuessRegistryPath(image string) (*nurl.URL, error) {
+
+	url, err := nurl.Parse(image)
+	if err != nil {
+		return url, microerror.Mask(err)
+	}
+
+	// Image string already has FQDN
+	if url.Hostname() != "" {
+		return nurl.Parse(image)
+	}
+
+	pathParts := strings.Split(strings.Trim(url.Path, "/"), "/")
+	if len(pathParts) == 2 {
+		// Image string is in the form organization/image, so we assume it is on docker hub
+		return nurl.Parse(fmt.Sprintf("https://registry.hub.docker.com/%s", image))
```

Comment:
This seems like a weird assumption to me. Do we make this assumption somewhere already?

Reply:
I agree, but I think it holds and I can't think of a more automatic way to resolve these names. We don't do this elsewhere because before we let docker handle this resolution when we pulled the image. This is for handling the three ways an image might be listed in the config:
- `some.domain.com/some/full/path` --> FQDN case
- `some/image` --> Docker hub case --> exactly two parts
- `someimage` --> Docker library case --> exactly one part

I don't think it's possible for `some/longer/path` to be a valid docker image unless it also specifies the host

Reply:
I found a much better way to do this today, will replace this